drivers/sensor/ina219/ina219.c

```diff
@@ -0,0 +1,315 @@
+		chan == SENSOR_CHAN_POWER)	{
+
+		rc = ina219_reg_read(dev, INA219_REG_POWER, &tmp);
+		if (rc) {
+			LOG_ERR("Error reading power register.");
+			return rc;
+		}
+		data->power = tmp;
+	}
+
+	if (chan == SENSOR_CHAN_ALL ||
+		chan == SENSOR_CHAN_CURRENT) {
+
+		rc = ina219_reg_read(dev, INA219_REG_CURRENT, &tmp);
+		if (rc) {
+			LOG_ERR("Error reading current register.");
+			return rc;
+		}
+		data->current = tmp;
+	}
+
+	return rc;
+}
+
+static int ina219_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	const struct ina219_config *cfg = dev->config;
+	struct ina219_data *data = dev->data;
+	double tmp;
+	int8_t sign = 1;
+
+	switch (chan) {
+	case SENSOR_CHAN_VOLTAGE:
+		tmp = data->v_bus * INA219_V_BUS_MUL;
+		sensor_value_from_double(val, tmp);
+		break;
+	case SENSOR_CHAN_POWER:
+		tmp = data->power * cfg->current_lsb * INA219_POWER_MUL * INA219_SI_MUL;
+		sensor_value_from_double(val, tmp);
+		break;
+	case SENSOR_CHAN_CURRENT:
+		if (INA219_SIGN_BIT(data->current)) {
+			data->current = ~data->current + 1;
+			sign = -1;
+		}
+		tmp = sign * data->current * cfg->current_lsb * INA219_SI_MUL;
+		sensor_value_from_double(val, tmp);
+		break;
+	default:
+		LOG_DBG("Channel not supported by device!");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int ina219_pm_ctrl(const struct device *dev, enum pm_device_action action)
+{
+	int rc = 0;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		return ina219_init(dev);
+	case PM_DEVICE_ACTION_SUSPEND:
+		reg_val = INA219_MODE_SLEEP;
+		break;
+	case PM_DEVICE_ACTION_TURN_OFF:
+		reg_val = INA219_MODE_OFF;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return ina219_reg_field_update(dev,
+				INA219_REG_CONF,
+				INA219_MODE_MASK,
+				reg_val);
+}
+#endif /* CONFIG_PM_DEVICE */
+
+static int ina219_init(const struct device *dev)
+{
+	const struct ina219_config *cfg = dev->config;
+	int rc;
+
+	if (!device_is_ready(cfg->bus)) {
+		LOG_ERR("Device not ready.");
+		return -ENODEV;
+	}
+
+	rc = ina219_reg_write(dev, INA219_REG_CONF, INA219_RST);
+	if (rc) {
+		LOG_ERR("Could not reset device.");
+		return rc;
+	}
+
+	rc = ina219_set_config(dev);
+	if (rc) {
+		LOG_ERR("Could not set configuration data.");
+		return rc;
+	}
+
+	rc = ina219_set_calib(dev);
+	if (rc) {
+		LOG_DBG("Could not set calibration data.");
+		return rc;
+	}
+
+	/* Set measurement delay */
+	rc = ina219_set_msr_delay(dev);
+	if (rc) {
+		LOG_ERR("Could not get measurement delay.");
+		return rc;
+	}
+
+	k_sleep(K_USEC(INA219_WAIT_STARTUP));
+
+	return 0;
+}
+
+static const struct sensor_driver_api ina219_api = {
+	.sample_fetch = ina219_sample_fetch,
+	.channel_get = ina219_channel_get,
+};
+
+
+#define INA219_INIT(n)						\
+	static struct ina219_data ina219_data_##n;		\
+								\
+	static const struct ina219_config ina219_config_##n = {	\
+		.bus = DEVICE_DT_GET(DT_INST_BUS(n)),		\
+		.i2c_addr = DT_INST_REG_ADDR(n),		\
```

Comment:
use `i2c_dt_spec`: https://github.com/zephyrproject-rtos/zephyr/pull/37359

Reply:
updated the driver accordingly.
I could also update the sgp40 and sht4x driver and add it to this pull request.
Would that be OK?

Reply:
better open another PR for that, would be a nice improvement :-)

Reply:
see: https://github.com/zephyrproject-rtos/zephyr/pull/37741